hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -2475,13 +2475,21 @@ public RSRpcServices getRSRpcServices() {
    */
   @Override
   public void abort(String reason, Throwable cause) {
+    synchronized (this) {
```

Comment:
An AtomicBoolean with CAS is enough?

Reply:
Okay. Used synchronized to keep the changes minimal. But CAS is good too, changed it.